### ISSUE TITLE
[issue]: Request for injection plugin support in WIMBOOT mode 

### ISSUE THREAD

**Issue by devdevadev** [ventoy/Ventoy#1834]
### Official FAQ

- [X] I have checked the official FAQ.

### Ventoy Version

1.0.79

### What about latest release

Yes. I have tried the latest release, but the bug still exist.

### BIOS Mode

Both

### Partition Style

MBR

### Disk Capacity

32GB

### Disk Manufacturer

Samsung

### What happened?

Hi....LongPanda

First lots of thanks for providing injection plugin feature in ventoy that really help for blind/Visually Impaired users....

I am currently enjoying injection plugin (windows_injection.7z) when installing Windows ISO in normal mode using Ventoy.  But today when I was booted WIN11 ISO in my another PC, then Ventoy does not booted to Windows Setup even after 1 hour. It stuck to a blinking cursor point.....So I decide to boot WIN11 ISO in WIMBOOT mode. When I booted to WIMBOOT mode it directly reach to Windows Setup screen and does not uses injection plugin (windows_injection.7z). 

I am a visually Imparied persion and use my own WIN_SMART_INSTALL.cmd within .7z package in order to automatic silent installation of Windows using injection plugin features....But here I become disappointed and I have to use Grubfm in order to install WIN11 ISO in automatic mode.

Can you please provide support of injection plugin in WIMBOOT mode also so we can enjoy Ventoy injection plugin in both normal mode and WIMBOOT mode......

Thanks & Regards....

**Comment by ventoy** [comment]
You can try this CI release: https://github.com/ventoy/Ventoy/actions/runs/3038203423
https://www.ventoy.net/en/doc_github_ci.html

**Comment by devdevadev** [comment]
Thanks a lot for your kind consideration....

Now I will be able to install WIN11 in my Macbook Air automatically and silently in WIMBOOT mode using injection plugin features.....

Love you LongPanda....

**Comment by devdevadev** [comment]
Just tested latest CI release.....

But Injection plugin still does not seems to work in WIMBOOT mode...!!!

Regards...

**Comment by steve6375** [comment]
![image](https://user-images.githubusercontent.com/1469350/189720083-a2faeee0-97f8-4b1e-aae5-5a97d665cf43.png)
Maybe vtoyjump is aborting and not running original winpeshl.exe ???

**Comment by steve6375** [comment]
![image](https://user-images.githubusercontent.com/1469350/189721922-fbe99225-308f-4234-a655-066a3baef659.png)

**Comment by ventoy** [comment]
> Just tested latest CI release.....
> 
> But Injection plugin still does not seems to work in WIMBOOT mode...!!!
> 
> Regards...

What is the phenomenon?  Did the setup dialog pop up? If the setup dialog pop up, press `Shift+F10` then run `type  X:\Windows\system32\ventoy.log`

**Comment by devdevadev** [comment]
Setup dialong popup instead of running my custom WIN_SMART_INSTALL.cmd batch file....

[ventoy.log](https://github.com/ventoy/Ventoy/files/9552554/ventoy.log)

**Comment by ventoy** [comment]
The log shows that Ventoy has decompressed `windows_injection.7z` and run `X:\VentoyAutoRun.bat`.
Please check that if there exist  `X:\VentoyAutoRun.log`

```
[2022/09/13 00:18:24.744] [1124] File <F:/ventoy/windows_injection.7z> exist
[2022/09/13 00:18:24.744] [1124] decompress injection archive F:/ventoy/windows_injection.7z...
[2022/09/13 00:18:24.744] [1124] DecompressInjectionArchive F:/ventoy/windows_injection.7z
[2022/09/13 00:18:24.760] [1124] Read MBR Ret:1 Size:512 code:0
[2022/09/13 00:18:24.760] [1124] MBR partition style
[2022/09/13 00:18:24.760] [1124] GetVentoyEfiPart StartSector: 57022440
[2022/09/13 00:18:24.760] [1124] Parse FAT fs...
[2022/09/13 00:18:24.760] [1124] CopyFileFromFatDisk (/ventoy/7z/64/7za.xz)==>(ventoy\7za.xz)
[2022/09/13 00:18:25.025] [1124] SaveBuffer2File <ventoy\7za.xz> len:425632
[2022/09/13 00:18:25.025] [1124] ReadWholeFile2Buf <ventoy\7za.xz>
[2022/09/13 00:18:25.025] [1124] Success read file size:425632
[2022/09/13 00:18:25.025] [1124] 7za.xz file size:425632
[2022/09/13 00:18:25.072] [1124] Decompress success 7za.xz(425632) ---> 7za.exe(1152000)
[2022/09/13 00:18:25.072] [1124] SaveBuffer2File <ventoy\7za.exe> len:1152000
[2022/09/13 00:18:25.103] [1124] extract inject to X:
[2022/09/13 00:18:25.103] [1124] cmdline:<ventoy\7za.exe x -y -aoa -oX:\ F:/ventoy/windows_injection.7z>
[2022/09/13 00:18:25.134] [1124] File <X:\VentoyAutoRun.bat> exist
[2022/09/13 00:18:25.134] [1124] X:\VentoyAutoRun.bat exist, now run it...
```

**Comment by devdevadev** [comment]
![Winpeshl](https://user-images.githubusercontent.com/20319541/189819852-9c006926-4714-4408-bf53-f6795e2b577f.jpg)

E:\ventoy\windows_injection.7z\VentoyAutoRun.bat -
================================================
@echo off
set ISOFILE=%1
set MNTPOINT=%2
echo ISOFILE=%ISOFILE%
echo MNTPOINT=%MNTPOINT%

Yes..you are right. Maybe VentoyAutoRun.bat is running in the background.
But I want to run my start_vtoy.bat using Winpeshl.ini on the top so I can
choose my desired Installation type and enter some user inputs before installation
start. I use following Winpeshl.ini and start_vtoy.bat for this purpose and it works
perfectly in normal mode but not working in WIMBOOT mode.


E:\ventoy\windows_injection.7z\Windows\System32\Winpeshl.ini -
============================================================
[LaunchApps]
%SystemRoot%\system32\wpeinit.exe
%SystemRoot%\system32\netstart.exe
x:\windows\system32\start_vtoy.bat


E:\ventoy\windows_injection.7z\Windows\System32\start_vtoy.bat -
===============================================================
@echo off
title Ventoy start_vtoy.bat
echo Running wpeinit.exe - please wait...
%cd%\wpeinit.exe
cls
ren %cd%\wpeinit.exe wpeinit.exe.old
cmd /q /c  "FOR %%i IN (C D E F G H I J K L N M O P Q R S T U V W X Y Z) DO IF EXIST %%i:\_ISO\WINDOWS\INSTALLS\CONFIGS\WIN_SMART_INSTALL.CMD  cmd /k %%i:\_ISO\WINDOWS\INSTALLS\CONFIGS\WIN_SMART_INSTALL.CMD"

**Comment by ventoy** [comment]
Please upload your injection archive here for test.

**Comment by devdevadev** [comment]
[windows_injection.zip](https://github.com/ventoy/Ventoy/files/9555268/windows_injection.zip)

**Comment by ventoy** [comment]
@devdevadev 
Try this CI release: https://github.com/ventoy/Ventoy/actions/runs/3045726482

**Comment by ventoy** [comment]
Fixed in 1.0.80
